src/libraries/icubmod/usbCamera/linux/V4L_camera.cpp

```diff
@@ -768,7 +768,7 @@ bool V4L_camera::getRawBuffer(unsigned char *buffer)
     mutex.wait();
     if(configured){
         imageProcess(param.raw_image);
-        memcpy(buffer, param.dst_image, param.dst_image_size);
+        memcpy(buffer, param.tmp_image, param.dst_image_size);
```

Comment:
Well, the idea is that the `getRawBuffer` has to return the image before the color reconstruction, while the `dst_image` is the buffer containing the rgb image.

Usually the raw images are stored in `param.raw_image` but for the leopard cameras a minimum processing is still required, and the result is stored in `param.tmp_image`

So before it was wrong to use the `dst_image` because ti was the rgb image.
Now it is ok for leopard... but it is probably still wrong for standard cameras, because it shall be `param.raw_image`.
I'll check. Thanks for the catch